openxr/examples/wgpu.rs

```diff
@@ -0,0 +1,645 @@
+    // like your rendering API might not be provided by the active runtime. APIs like OpenGL don't
+    // have universal support.
+    assert!(available_extensions.khr_vulkan_enable2);
+
+    // Initialize OpenXR with the extensions we've found!
+    let mut enabled_extensions = xr::ExtensionSet::default();
+    enabled_extensions.khr_vulkan_enable2 = true;
+    #[cfg(target_os = "android")]
+    {
+        enabled_extensions.khr_android_create_instance = true;
+    }
+    let xr_instance = xr_entry
+        .create_instance(
+            &xr::ApplicationInfo {
+                application_name: "openxrs example",
+                application_version: 0,
+                engine_name: "openxrs example",
+                engine_version: 0,
+            },
+            &enabled_extensions,
+            &[],
+        )
+        .unwrap();
+    let instance_props = xr_instance.properties().unwrap();
+    println!(
+        "loaded OpenXR runtime: {} {}",
+        instance_props.runtime_name, instance_props.runtime_version
+    );
+
+    // Request a form factor from the device (HMD, Handheld, etc.)
+    let system = xr_instance
+        .system(xr::FormFactor::HEAD_MOUNTED_DISPLAY)
+        .unwrap();
+
+    // Check what blend mode is valid for this device (opaque vs transparent displays). We'll just
+    // take the first one available!
+    let environment_blend_mode = xr_instance
+        .enumerate_environment_blend_modes(system, VIEW_TYPE)
+        .unwrap()[0];
+
+    // OpenXR wants to ensure apps are using the correct graphics card and Vulkan features and
+    // extensions, so the instance and device MUST be set up before Instance::create_session.
+
+    let vk_target_version = vk::make_api_version(0, 1, 1, 0); // Vulkan 1.1 guarantees multiview support
+    let vk_target_version_xr = xr::Version::new(1, 1, 0);
+
+    let reqs = xr_instance
+        .graphics_requirements::<xr::Vulkan>(system)
+        .unwrap();
+
+    if vk_target_version_xr < reqs.min_api_version_supported
+        || vk_target_version_xr.major() > reqs.max_api_version_supported.major()
+    {
+        panic!(
+            "OpenXR runtime requires Vulkan version > {}, < {}.0.0",
+            reqs.min_api_version_supported,
+            reqs.max_api_version_supported.major() + 1
+        );
+    }
+
+    let (_wgpu_instance, _adapter, device, queue, session, mut frame_wait, mut frame_stream) = unsafe {
+        let vk_entry = ash::Entry::load().unwrap();
+
+        let vk_app_info = vk::ApplicationInfo::builder()
+            .application_version(0)
+            .engine_version(0)
+            .api_version(vk_target_version);
+
+        let vk_instance = {
+            let vk_instance = xr_instance
+                .create_vulkan_instance(
+                    system,
+                    std::mem::transmute(vk_entry.static_fn().get_instance_proc_addr),
+                    &vk::InstanceCreateInfo::builder().application_info(&vk_app_info) as *const _
```

Comment:
You should account for `HalVulkan::required_extensions` when creationg the Vulkan instance (enabled_extension_names)

Reply:
I just updated this to check for extensions on the device and instance. I was originally copying the Vulkan example, which didn't seam to check those. But I think your right and I should be checking them